### PR TITLE
Move Build Server integration to plugin

### PR DESCRIPTION
**Pull request by ivangrek**
ref to #8766

## Proposed changes

- Move Build Server integration to plugin to transfer to a separate repo

### Before

image

### After

image

## Test methodology 

- Manually
- Visually

## Test environment(s) 

- Git Extensions 33.33.33
- Build b69f89ef38d59ee8657f8df2451f2ceb9cdc9f59 (Dirty)
- Git 2.30.0.windows.2
- Microsoft Windows NT 10.0.19041.0
- .NET Framework 4.8.4300.0
- DPI 96dpi (no scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by gerhardol — COMMENTED** [review]
Build servers is a feature that is very tightly built into GE. I do not see a reason to separate that from the core.
Where is the translation?

**Comment by ivangrek** [comment]
> Build servers is a feature that is very tightly built into GE. I do not see a reason to separate that from the core.

Make better.

> Where is the translation?

I will restore.

**Comment by RussKie** [comment]
> Build servers is a feature that is very tightly built into GE. I do not see a reason to separate that from the core.

I respectfully disagree on this point. Whilst I do use the AppVeyor build server integration for GE and the AzDO integration for dotnet/winforms, I don't use it for other projects. 
And whilst delivering these plugins inbox is convenient for users, it is probably more inconvenience when the plugins don't work well, and the users can't get an update because of our slow release cadence.

File: GitExtensions.sln [hunk 1]
```diff
@@ -218,6 +218,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "UI.IntegrationTests", "Inte
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "setup", "setup", "{2786EE4D-2C03-4622-8800-8A66ECEF2AA5}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "GitExtensions.Plugins.BuildServer", "Plugins\AppVeorBuldReport\GitExtensions.Plugins.BuildServer.csproj", "{BED49EEA-19A3-4692-B402-95184D4D38C7}"
```
Comment (RussKie) [inline GitExtensions.sln:221]:
```suggestion
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "GitExtensions.Plugins.BuildServer", "Plugins\AppVeyorBuldReport\GitExtensions.Plugins.BuildServer.csproj", "{BED49EEA-19A3-4692-B402-95184D4D38C7}"
```

File: Plugins/AppVeorBuldReport/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+﻿using System.Reflection;
+
+[assembly: AssemblyDescription("GitExtensions plugin for integration with Build Server")]
```
Comment (RussKie) [inline Plugins/AppVeorBuldReport/Properties/AssemblyInfo.cs:3]:
```suggestion
[assembly: AssemblyDescription("GitExtensions plugin for AppVeyor integration")]
```

File: Plugins/AppVeorBuldReport/Settings.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        public string Name { get; }
+
+        public string Caption { get; }
+
+        public BuildServerIntegrationSettingsPage Control { get; }
+
+        public ISettingControlBinding CreateControlBinding()
+        {
+            return this;
+        }
```
Comment (RussKie) [inline Plugins/AppVeorBuldReport/Settings.cs:23]: [nit]
```suggestion
        public ISettingControlBinding CreateControlBinding() => this;
```
ditto below

**Review by RussKie — COMMENTED** [review]
👀 

What's the vision for other build server integrations?

File: Plugins/AppVeorBuldReport/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (gerhardol) [reply Plugins/AppVeorBuldReport/Properties/AssemblyInfo.cs:3]: No, this is in the build server integration, some parts are apparently needed for AppVeyor here - a separate refactoring

Comment (ivangrek) [reply Plugins/AppVeorBuldReport/Properties/AssemblyInfo.cs:3]: "AppVeorBuldReport" invalid folder name. Plugin for all types.

**Comment by ivangrek** [comment]
> What's the vision for other build server integrations?

Plugin for all current types of build servers.

image

**Comment by ivangrek** [comment]
@drewnoakes, please annotate plugins too. My changes is blocked now.

**Comment by drewnoakes** [comment]
> @drewnoakes, please annotate plugins too. My changes is blocked now.

Do you mean the individual plugins projects? I don't know when I would be able to get to that, as it will take a while. And annotating all of those is likely to increase your conflicts rather than reduce them. Am I missing something?

**Comment by ivangrek** [comment]
> Do you mean the individual plugins projects? I don't know when I would be able to get to that, as it will take a while. And annotating all of those is likely to increase your conflicts rather than reduce them. Am I missing something?

Everything turned out to be easier than I thought.

**Pull request closed without merging**